Source/DafnyStandardLibraries/src/Std_Concurrent/Std_Concurrent.go

```diff
@@ -58,79 +68,152 @@ func (_this *MutableMap) ParentTraits_() []*_dafny.TraitID {
 }
 var _ _dafny.TraitOffspring = &MutableMap{}
 
-func (_this *MutableMap) Ctor__() {
+func (_this *MutableMap) Ctor__(bytesKeys bool) {
   {
+    _this.bytesKeys = bytesKeys
   }
 }
+
+func dafnyBytesToString(b interface{}) string {
+  return string(_dafny.ToByteArray(b.(_dafny.Sequence)))
+}
+func dafnyBytesFromString(b interface{}) _dafny.Sequence {
+  return _dafny.SeqOfBytes([]byte(b.(string)))
+}
+
 func (_this *MutableMap) Keys() _dafny.Set {
   {
-    _this.mu.Lock()
-    defer _this.mu.Unlock()
+    if _this.bytesKeys {
+      keys := make([]interface{}, 0)
+
+      _this.goInternal.Range(func(key, value interface{}) bool {
+        keys = append(keys, dafnyBytesFromString(key))
+        return true
+      })
+  
+      return _dafny.SetOf(keys[:]...)
+    } else {
+      _this.mu.Lock()
+      defer _this.mu.Unlock()
 
-    return _this.Internal.Keys()
+      return _this.dafnyInternal.Keys()
+    }
   }
 }
 func (_this *MutableMap) HasKey(k interface{}) bool {
   {
-    _this.mu.Lock()
-    defer _this.mu.Unlock()
+    if _this.bytesKeys {
+      _, ok := _this.goInternal.Load(dafnyBytesToString(k))
+      return ok
+    } else {
+      _this.mu.Lock()
+      defer _this.mu.Unlock()
 
-    return _this.Internal.Contains(k)
+      return _this.dafnyInternal.Contains(k)
+    }
   }
 }
 func (_this *MutableMap) Values() _dafny.Set {
   {
-    _this.mu.Lock()
-    defer _this.mu.Unlock()
+    if _this.bytesKeys {
+      values := make([]interface{}, 0)
 
-    return _this.Internal.Values()
+      _this.goInternal.Range(func(key, value interface{}) bool {
+        values = append(values, value)
+        return true
+      })
+
+      return _dafny.SetOf(values[:]...)
+    } else {
+      _this.mu.Lock()
+      defer _this.mu.Unlock()
+
+      return _this.dafnyInternal.Values()
+    }
   }
 }
 func (_this *MutableMap) Items() _dafny.Set {
   {
-    _this.mu.Lock()
-    defer _this.mu.Unlock()
+    if _this.bytesKeys {
+      items := make([]interface{}, 0)
+
+      _this.goInternal.Range(func(key, value interface{}) bool {
+        items = append(items, _dafny.TupleOf(dafnyBytesFromString(key), value))
+        return true
+      })
 
-    return _this.Internal.Items()
+      return _dafny.SetOf(items[:]...)
+    } else {
+      _this.mu.Lock()
+      defer _this.mu.Unlock()
+
+      return _this.dafnyInternal.Items()
+    }
   }
 }
 func (_this *MutableMap) Get(k interface{}) Std_Wrappers.Option {
   {
-    _this.mu.Lock()
-    defer _this.mu.Unlock()
-
-    value, ok := _this.Internal.Find(k)
-    if ok {
-      return Std_Wrappers.Companion_Option_.Create_Some_(value)
+    if _this.bytesKeys {
+      value, ok := _this.goInternal.Load(dafnyBytesToString(k))
+      if ok {
+        return Std_Wrappers.Companion_Option_.Create_Some_(value)
+      } else {
+        return Std_Wrappers.Companion_Option_.Create_None_()
+      }
```

Comment:
Do you test this?

Reply:
Good catch, added.

Unfortunate that we're only measuring Dafny code coverage given we have extern code to test coverage of too.